### ISSUE TITLE
Remove global julia registry

### ISSUE THREAD

**Issue by haampie** [eth-cscs/production#2407]
Currently we have a site-wide julia registry when using `module load Julia`, since we're setting `JULIA_DEPOT_PATH`.

The upside of this is that everybody has the same registry, the downside is that users can't update the registry. This means that users can't get new packages after Julia has been installed on the system, other than unsetting JULIA_DEPOT_PATH.

Since Julia packages are often fast release cycles, it'd be great if any user can update the registry for themselves (that is: run `] up`).

(assigning @omlins so that he gets notified ;) )

**Comment by haampie** [comment]
Okay, the registry is indeed under `LOAD_PATH`, and julia searches for the first available registry through the list of directories. 

This means that if the user doesn't have a registry in e.g. `~/.julia/1.6.0/daint-gpu` it will use the registry downloaded by easybuild when julia was installed, and that's read-only for the user, so it's not updated:

```
$ module load daint-gpu Julia
$ julia
(@1.6.0-daint-gpu) pkg> up
  No Changes to `~/.julia/1.6.0/daint-gpu/environments/1.6.0-daint-gpu/Project.toml`
  No Changes to `~/.julia/1.6.0/daint-gpu/environments/1.6.0-daint-gpu/Manifest.toml`
```

However, if you disable the startup script like this:

```
$ unset JULIA_PKG_SERVER
$ julia --startup=no
(@1.6.0-daint-gpu) pkg> up
  Installing known registries into `~/.julia/1.6.0/daint-gpu`
     Cloning registry from "https://github.com/JuliaRegistries/General.git"
...

[hundreds of packages are upgraded...]
```
it will download the latest registry in your personal home dir, and later it will always use this registry, not the easybuild one, since it's first to be found in the `LOAD_PATH`. So this is a workaround to get new packages, but I think it's better to make this the default. What do you think @omlins?

**Comment by omlins** [comment]
Good point. I totally agree that each user should manage his own registry in the user space. However, just disabling the startup by default will affect the current way of working of the modules. I will search for a solution...
